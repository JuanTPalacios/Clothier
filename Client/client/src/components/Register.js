import React from 'react'
import { useState } from 'react';
import { Link } from "react-router-dom";



export default function Register({setAuthenticated}) {
  const [registryStep, setRegistryStep] = useState({
    step: 1,
  });
  switch (registryStep) {
    case 2:
      return (
      <div>
        Register Form Step 2
        <button className="button" type='click' onClick={() => {setRegistryStep(3)}} > Next </button>
      </div>)
    case 3:
      return (
      <div>
        Register Form Step 3
        <button className="button" type='click' onClick={() => {setRegistryStep(4)}} > Next </button>
      </div>)
    case 4:
      return (
      <div>
        Registration succesful, click to go to home (here auth must me set to true)
        <Link to="/">
          <button className="button" type='click' onClick={() => {setAuthenticated(true)}}> Next </button>
        </Link>
      </div>)
    default:
      return (
        <div>
        Register Form Step 1
        <button className="button" type='click' onClick={() => {setRegistryStep(2)}} > Next </button>
      </div>
      )
  };
};
