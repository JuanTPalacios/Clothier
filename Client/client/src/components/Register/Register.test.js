import {render, screen} from '@testing-library/react';
import Register from './Register';

describe('Register Page', () => {
  test('should match the snapshot', () => {
    const { container } = render(<Register />);
    expect(container.firstChild).toMatchSnapshot();
  })
})