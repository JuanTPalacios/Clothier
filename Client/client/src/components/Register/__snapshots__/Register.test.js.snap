// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page should match the snapshot 1`] = `
<div
  class="login-page is-flex is-justify-content-center"
>
  <div
    class="login-window box"
  >
    <h1
      class="title is-2"
    >
      Welcome to Clothier!
    </h1>
    <h1
      class="title is-5"
    >
      Please follow these steps to begin:
    </h1>
    <div
      class="field"
    >
      <form
        class="form"
      >
        <input
          class="mt-1 input"
          name="email"
          placeholder="name@mail.com"
          required=""
          type="email"
          value=""
        />
        <input
          class="mt-1 input"
          name="username"
          placeholder="Username"
          required=""
          type="text"
          value=""
        />
        <input
          class="mt-1 input"
          name="password"
          placeholder="password"
          required=""
          type="password"
          value=""
        />
        <input
          class="mt-1 input"
          name="firstName"
          placeholder="Name"
          required=""
          type="text"
          value=""
        />
        <input
          class="mt-1 input"
          name="lastName"
          placeholder="Last name"
          type="text"
          value=""
        />
        <button
          class="mt-4 button is-primary is-rounded"
          type="submit"
        >
           Next 
        </button>
      </form>
    </div>
  </div>
</div>
`;
